contracts/BondCurve.aes

```diff
@@ -0,0 +1,53 @@
+@compiler >=4.2
+
+include "Frac.aes"
+
+contract BondCurve =
+
+ // token : TokenContract
+  record state = { init_buy_price : Frac.frac
+                 , init_sell_price : Frac.frac
+                 , alpha: Frac.frac }
+
+  stateful entrypoint init() =
+    // { total_supply = 0, init_buy_price = 1, init_sell_price = 0, alpha = 1 }
+    { init_buy_price = Frac.from_int(1), init_sell_price = Frac.from_int(0), alpha = Frac.from_int(1) }
+    
+  // Returns the current buy price
+  entrypoint buy_price(total_supply: Frac.frac) : Frac.frac =
+    buy_curve(total_supply)
+  
+  // Returns the current sell price
+  entrypoint sell_price(total_supply: Frac.frac) : Frac.frac =
+    sell_curve(total_supply)
+  
+  // Returns the amount of fungible tokens to be received
+  // when buying from the curve for the passed value of AE tokens.
+  // entrypoint calculate_buy_return(amount: Frac.frac) : Frac.frac = 0
+    // delta = b^2 + 2a*((a * c^2/2 + b * c) + AE)
+    // n = (-b + sqrt(delta)) / a
+    // Requires SQRT and approximation which is impossible rn
```

Comment:
This is here just for reference (so we have it in the history if at any point we have `sqrt` library in Sophia and we decide to simplify the calculations) and it will be removed in the next commits.